launcher/minecraft/mod/ModFolderModel.cpp

```diff
@@ -196,7 +196,10 @@ Task* ModFolderModel::createParseTask(Resource& resource)
 bool ModFolderModel::uninstallMod(const QString& filename, bool preserve_metadata)
 {
     for (auto mod : allMods()) {
-        if (mod->fileinfo().fileName() == filename) {
+        auto modfilename = mod->fileinfo().fileName();
+        if (!mod->enabled() && modfilename.endsWith(".disabled"))
+            modfilename.chop(9);
+        if (modfilename == filename) {
```

Comment:
This could definitely be cleaner. Firstly, the `.disabled` check is redundant - secondly, there should be a function or field in `Resource` for this